io.openems.backend.metadata.odoo/src/io/openems/backend/metadata/odoo/EdgeCache.java

```diff
@@ -117,12 +115,12 @@ public synchronized MyEdge getEdgeForApikey(String apikey) {
 	}
 
 	/**
-	 * Gets all Edges as an unmodifiable Collection.
+	 * Returns a sequential stream with this {@link EdgeCache} as its source.
 	 *
-	 * @return a collection of Edges
+	 * @return a sequential {@link Stream} of the {@link MyEdge} in this cache.
 	 */
-	public Collection<Edge> getAllEdges() {
-		return Collections.unmodifiableCollection(this.edgeIdToEdge.values());
+	public Stream<MyEdge> stream() {
+		return this.edgeIdToEdge.values().stream();
```

Comment:
I tend to prefer returning a List instead of a Stream in these situations, as it allows more possibilities for the consumer of the method.